### PR TITLE
Multiple ndb improvements

### PR DESCRIPTION
**Pull request by mlschroe**

### CODE REVIEW

File: lib/backend/ndb/rpmxdb.c [hunk 1]
```diff
@@ -228,16 +265,10 @@ static int rpmxdbReadHeader(rpmxdb xdb)
     if (!slotnpages || !pagesize || stb.st_size % pagesize != 0)
 	return RPMRC_FAIL;
     xdb->pagesize = pagesize;
+    xdb->mapflags = xdb->rdonly ? PROT_READ : PROT_READ | PROT_WRITE, MAP_SHARED;
```
Comment (pmatilai) [inline lib/backend/ndb/rpmxdb.c]: This generates a warning:
```
backend/ndb/rpmxdb.c: In function ‘rpmxdbReadHeader’:
backend/ndb/rpmxdb.c:268:69: warning: right-hand operand of comma expression has no effect [-Wunused-value]
  268 |     xdb->mapflags = xdb->rdonly ? PROT_READ : PROT_READ | PROT_WRITE, MAP_SHARED;
      |    
```

I assume the ',' is supposed to be a '|' instead :)

**Comment by mlschroe** [comment]
It just needs to be removed, the MAP_SHARED was the next argument of the mmap call. Force-pushed new version.

**Comment by pmatilai** [comment]
Roger that.

**Comment by pmatilai** [comment]
@mlschroe - given the recent resurgence in ndb activity, just thought to mention this so we're clear: removing the EXPERIMENTAL tag on ndb is going to be up to you and I'm going to trust your judgement on that.

**Pull request merged**